tests/network/l2_bridge/libl2bridge.py

```
@@ -236,18 +258,82 @@ def hot_plug_interface_and_set_address(
+    raise VMInterfaceStatusNotFoundError(f"No interface with MAC {expected_mac} found inside VM {vm.name}")
+
+
+def _read_guest_ipv4(vm: VirtualMachine, interface_name: str) -> ipaddress.IPv4Address:
+    cmd = f"ip -j -4 addr show {interface_name}"
+    output = vm_console_run_commands(vm=vm, commands=[cmd], timeout=30)
+    iface_info = json.loads(output[cmd][1])
+    if iface_info and "addr_info" in iface_info[0]:
+        for addr in iface_info[0]["addr_info"]:
+            if addr["family"] == "inet":
+                return ipaddress.IPv4Address(address=addr["local"])
+    raise IpNotFound(f"No IPv4 address found on {interface_name} in VM {vm.name}")
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)